Index: node_export.module
===================================================================
--- node_export.module	(revision 172)
+++ node_export.module	(working copy)
@@ -1146,7 +1146,7 @@
 
     if (!empty($query)) {
       watchdog('node_export', 'kept existing managed file at uri "%uri"', array('%uri' => $file->uri), WATCHDOG_NOTICE);
-      $file = file_load(array_shift($query));
+      $file = (object) array_merge((array) $file, (array) file_load(array_shift($query)));
     }
 
     $file = file_save($file);
